File: sast-python2/case/completeness/single_app_tracing/alias/alias_005_T.py
# -*- coding: utf-8 -*-
# evaluation information start
# real case = true
# evaluation item = 完整度->单应用跟踪完整度->别名
# scene introduction = 列表元素别名
# level = 2
# bind_url = completeness/single_app_tracing/alias/alias_005_T
# evaluation information end
import os


def alias_005_T(taint_src):
    a = [u'_', u'_']
    b = a  # 别名
    b[0] = taint_src  # 修改列表元素
    taint_sink(a[0])  # 应被污染

def taint_sink(o):
    os.system(o)

if __name__ == u"__main__":
    taint_src = u"taint_src_value"
    alias_005_T(taint_src)

File: sast-js/case/completeness/single_app_tracing/alias/alias_005_T.js
// evaluation information start
// real case = true
// evaluation item = 完整度->单应用跟踪完整度->别名
// scene introduction = 解构参数
// level = 2
// bind_url = completeness/single_app_tracing/alias/alias_005_T
// date = 2025-12-18 06:34:45
// evaluation information end

const { execSync } = require("child_process");

function alias_005_T(__taint_src) {
  const arr = [__taint_src, "save_value"];

  const [a, b] = arr;

  __taint_sink(a);
}

function __taint_sink(o) {
  execSync(o);
}

const taint_src = "taint_src_value";

alias_005_T(taint_src);
